analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ExplodedNode.cs

```diff
@@ -62,5 +62,10 @@ private ExplodedNode(BasicBlock block, IOperationWrapperSonar[] operations, int
             other is not null
             && other.programPointHash == programPointHash
             && other.State.Equals(State);
+
+        public override string ToString() =>
```

Comment:
For debugging, newlines are not particular useful. Consider to (also) have a [DebuggerDisplay] attribute that has a single line representation.

Reply:
Hi @Corniel, thanks for the suggestion. We've considered the `[DebuggerDisplay]` option and went for the `ToString`. While the `ExplodedNode` itself doesn't really need the new line, the `ProgramState` does.

The intended usage is to add `Console.WriteLine(current.ToString())` locally here, as the flows can be often tricky to follow and understand:
https://github.com/SonarSource/sonar-dotnet/blob/3f1f4cdbebba4d87acd571579389bb7d8324a9ed/analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs#L50-L62

Reply:
You could have a debugger display to, that does `ToString().Replace(Environement.NewLine, "; "). For Console.WriteLine, new lines indeed have added value.

Reply:
We'll see if that's needed later. For now, I'm a bit afraid that the state will be too long to read in a single line anyway based on the experience with the old engine.

Reply:
But I like the `Replace` idea if we'll need taht